dictionaries/CallMap.php

```diff
@@ -15643,46 +15643,46 @@
+'XMLWriter::endDtdAttlist' => ['bool'],
+'XMLWriter::endDtdElement' => ['bool'],
+'XMLWriter::endDtdEntity' => ['bool'],
 'XMLWriter::endElement' => ['bool'],
-'XMLWriter::endPI' => ['bool'],
-'XMLWriter::flush' => ['', 'empty='=>'bool', 'xmlwriter='=>''],
+'XMLWriter::endPi' => ['bool'],
+'XMLWriter::flush' => ['string|int|false', 'empty='=>'bool'],
 'XMLWriter::fullEndElement' => ['bool'],
 'XMLWriter::openMemory' => ['bool'],
-'XMLWriter::openURI' => ['bool', 'uri'=>'string'],
-'XMLWriter::outputMemory' => ['string', 'flush='=>'bool', 'xmlwriter='=>''],
-'XMLWriter::setIndent' => ['bool', 'indent'=>'bool'],
-'XMLWriter::setIndentString' => ['bool', 'indentstring'=>'string'],
+'XMLWriter::openUri' => ['bool', 'uri'=>'string'],
+'XMLWriter::outputMemory' => ['string', 'flush='=>'bool'],
+'XMLWriter::setIndent' => ['bool', 'enable'=>'bool'],
+'XMLWriter::setIndentString' => ['bool', 'indentation'=>'string'],
 'XMLWriter::startAttribute' => ['bool', 'name'=>'string'],
-'XMLWriter::startAttributeNS' => ['bool', 'prefix'=>'string', 'name'=>'string', 'uri'=>'string'],
-'XMLWriter::startCData' => ['bool'],
+'XMLWriter::startAttributeNs' => ['bool', 'prefix'=>'string', 'name'=>'string', 'namespace'=>'?string'],
+'XMLWriter::startCdata' => ['bool'],
 'XMLWriter::startComment' => ['bool'],
-'XMLWriter::startDocument' => ['bool', 'version='=>'string', 'encoding='=>'string', 'standalone='=>'string'],
-'XMLWriter::startDTD' => ['bool', 'qualifiedname'=>'string', 'publicid='=>'string', 'systemid='=>'string'],
-'XMLWriter::startDTDAttlist' => ['bool', 'name'=>'string'],
-'XMLWriter::startDTDElement' => ['bool', 'qualifiedname'=>'string'],
-'XMLWriter::startDTDEntity' => ['bool', 'name'=>'string', 'isparam'=>'bool'],
+'XMLWriter::startDocument' => ['bool', 'version='=>'?string', 'encoding='=>'?string', 'standalone='=>'?string'],
+'XMLWriter::startDtd' => ['bool', 'qualifiedName'=>'string', 'publicId='=>'?string', 'systemId='=>'?string'],
+'XMLWriter::startDtdAttlist' => ['bool', 'name'=>'string'],
+'XMLWriter::startDtdElement' => ['bool', 'qualifiedName'=>'string'],
+'XMLWriter::startDtdEntity' => ['bool', 'name'=>'string', 'isParam'=>'bool'],
 'XMLWriter::startElement' => ['bool', 'name'=>'string'],
-'XMLWriter::startElementNS' => ['bool', 'prefix'=>'string', 'name'=>'string', 'uri'=>'string'],
-'XMLWriter::startPI' => ['bool', 'target'=>'string'],
+'XMLWriter::startElementNs' => ['bool', 'prefix'=>'?string', 'name'=>'string', 'namespace'=>'?string'],
+'XMLWriter::startPi' => ['bool', 'target'=>'string'],
 'XMLWriter::text' => ['bool', 'content'=>'string'],
 'XMLWriter::writeAttribute' => ['bool', 'name'=>'string', 'value'=>'string'],
-'XMLWriter::writeAttributeNS' => ['bool', 'prefix'=>'string', 'name'=>'string', 'uri'=>'string', 'content'=>'string'],
-'XMLWriter::writeCData' => ['bool', 'content'=>'string'],
+'XMLWriter::writeAttributeNs' => ['bool', 'prefix'=>'string', 'name'=>'string', 'namespace'=>'string', 'value'=>'string'],
```

Comment:
The types here should probably match `startAttributeNs()`, shouldn't they?

Reply:
You're right that `namespace` should be nullable; I fixed that now. For the prefix, the same thing holds as for `startAttributeNs`: it is only nullable as of PHP 8.0. Although it is not mentioned in the manual, I found this confirmed in the corresponding [bug report](https://bugs.php.net/bug.php?id=79344) and [commit in PHP](http://git.php.net/?p=php-src.git;a=commit;h=aa754ba85e8241c0e59627c359d2c002a5caf8db).